File: chroma-league/src/main/java/net/booone/chromaleague/hud/colors/BackgroundBreathingColor.java
package net.booone.chromaleague.hud.colors;

import net.booone.chromaleague.hud.parts.Background;
import net.booone.razersdk.color.BreathingColor;
import net.booone.razersdk.color.StaticColor;

public class BackgroundBreathingColor extends BreathingColor {

    public BackgroundBreathingColor(StaticColor upColor) {
        super(upColor, Background.DEFAULT_BACKGROUND_COLOR);
    }

    public BackgroundBreathingColor(StaticColor upColor, int steps) {
        super(upColor, Background.DEFAULT_BACKGROUND_COLOR, steps);
    }

    public BackgroundBreathingColor(StaticColor upColor, int steps, boolean startUpDirection) {
        super(upColor, Background.DEFAULT_BACKGROUND_COLOR, steps, startUpDirection);
    }
}

File: src/hud/colors/BackgroundBreathingColor.ts
import {BreathingColor} from '../../razer-sdk/color/BreathingColor.js';
import {StaticColor} from '../../razer-sdk/color/StaticColor.js';
import {DEFAULT_BACKGROUND_COLOR} from '../parts/BackgroundConstants.js';

export class BackgroundBreathingColor extends BreathingColor {
    constructor(upColor: StaticColor, steps?: number, startUpDirection?: boolean) {
        if (steps !== undefined && startUpDirection !== undefined) {
            super(upColor, DEFAULT_BACKGROUND_COLOR, steps, startUpDirection);
        } else if (steps !== undefined) {
            super(upColor, DEFAULT_BACKGROUND_COLOR, steps);
        } else {
            super(upColor, DEFAULT_BACKGROUND_COLOR);
        }
    }
}
